avionics/envs/x86/include/stdio_controller.hpp

```diff
@@ -91,28 +139,57 @@ class StdIoController {
      * @return Read value.
      */
     static int requestAnalogRead(uint8_t const pin_id) {
-        run_input_ = false; // input_ will now run at most once more
-        putPacket('a', &pin_id,
-                  1);  // ensures that at least one more SIM packet exists
-        input_.join(); // so this won't block
+        uint8_t packet_id = static_cast<uint8_t>(PacketIds::analog_read);
+        BlockingRequest restart(packet_id, &pin_id, 1);
 
-        while (istreams_['a'].size() == 0) {
-            extractPacket();
-        }
-        int val = istreams_['a'].front();
-        istreams_['a'].pop();
+        int val = istreams_[packet_id].front();
+        istreams_[packet_id].pop();
         val *= 256;
-        val += istreams_['a'].front();
-        istreams_['a'].pop();
+        val += istreams_[packet_id].front();
+        istreams_[packet_id].pop();
 
-        // restart input loop
-        run_input_ = true;
-        std::thread input{&StdIoController::inputLoop};
-        input_ = std::move(input);
         return val;
     }
 
+    /**
+     * @brief Corresponds to Request Sensor Read packet in Confluence spec.
+     * @param sensor_id ID of the desired sensor.
+     * @param num_floats the number of floats to read.
+     * @return vector of sensor measurements.
+     */
+    static std::vector<float> requestSensorRead(uint8_t sensor_id,
+                                                std::size_t num_floats) {
+        uint8_t packet_id = static_cast<uint8_t>(PacketIds::sensor_read);
+        BlockingRequest restart(packet_id, &sensor_id, 1);
+
+        std::vector<float> sensor_data;
+        for (std::size_t f = 0; f < num_floats; f++) {
+            uint8_t packetData[FLOAT_SIZE];
+            for (std::size_t i = 0; i < FLOAT_SIZE; i++) {
+                packetData[i] = istreams_[packet_id].front();
+                istreams_[packet_id].pop();
+            }
+            sensor_data.push_back(charsToFloat(packetData));
+        }
+
+        return sensor_data;
+    }
+
   private:
+    /**
+     * @brief reinterprets an array of 4 chars to a floating point
+     * number.
+     * @param data: an array of 4 chars.
+     * @return the floating point representation
+     */
+
+    // NOTE Be mindful of endianess
```

Comment:
nitpicking but I believe that automatically generated documentation assumes that the comment block comes immediately before the function definition (not that we're generating documentation) - maybe move this comment inside

although for that matter, this should be clarified to say that ground station is taking care of endianness